src/models/loot_table.model.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    AllowNull,
+    AutoIncrement,
+    Column,
+    CreatedAt,
+    DataType,
+    PrimaryKey,
+    Table,
+    Model,
+    UpdatedAt
+} from 'sequelize-typescript';
+
+import { Floor } from '../models/floor.model.js'
+@Table({
+    freezeTableName: true,
+    tableName: 'loot_table'
+})
+export class LootTable extends Model<LootTable> {
+    @AutoIncrement
+    @AllowNull(false)
+    @PrimaryKey
+    @Column(DataType.INTEGER)
+    id: number;
+
+    @AllowNull(false)
+    @Column(DataType.ENUM('accessory_upgrade', 'armor_upgrade', 'bracelet', 'chest', 
+     'earrings', 'feet', 'gloves', 'head', 'legs', 'necklace', 'ring', 'offhand'))
+    piece_type: string;
+
+    @AllowNull(false)
+    @Column(DataType.STRING)
+    name_of_gear: string;
+
+        static associate(models) {
+            LootTable.belongsTo(models.floor);
+            models.floor.hasMany(LootTable);
+        }
+
+    @AllowNull(false)
+    @Column(DataType.STRING) 
```

Comment:
This needs @ForeignKey()

Reply:
@foreignkey() added but being hit with the red squiggles, will comment back and confirm resolve once this goes away